crates/ruff/src/noqa.rs

```diff
@@ -38,32 +40,93 @@ pub(crate) enum Directive<'a> {
 impl<'a> Directive<'a> {
     /// Extract the noqa `Directive` from a line of Python source code.
     pub(crate) fn try_extract(text: &'a str, offset: TextSize) -> Option<Self> {
-        let caps = NOQA_LINE_REGEX.captures(text)?;
-        match (caps.name("noqa"), caps.name("codes")) {
-            (Some(noqa), Some(codes)) => {
-                let codes = codes
-                    .as_str()
-                    .split(|c: char| c.is_whitespace() || c == ',')
-                    .map(str::trim)
-                    .filter(|code| !code.is_empty())
-                    .collect_vec();
-                if codes.is_empty() {
-                    warn!("Expected rule codes on `noqa` directive: \"{text}\"");
+        for mat in NOQA_MATCHER.find_iter(text) {
+            let noqa_literal_start = mat.start();
+
+            // Determine the start of the comment.
+            let mut comment_start = noqa_literal_start;
+
+            // Trim any whitespace between the `#` character and the `noqa` literal.
+            comment_start -= text[..comment_start].len() - text[..comment_start].trim_end().len();
```

Comment:
isn't this just the trimmed length? 
```suggestion
            comment_start = text[..comment_start].trim_end().len();
```